### PR TITLE
Update for Swift 4.0

### PR DESCRIPTION
**Pull request by paulw11**

### CODE REVIEW

**Comment by p2** [comment]
Cool, thanks! Can you please also update the `.swift-version` and `.travis.yml` files, and use tabs instead of spaces in the code files?

**Comment by paulw11** [comment]
Yeah, I am still working through it; I had to go do some other things.

I actually didn't touch most of that.  Perhaps the Xcode "fix this for Swift 4" has put spaces in.

I did change the lines near the #if because there is a bug in Xcode when it compiles for the migration tool it doesn't honour any -D in the build config, so I was trying to work around that.  In the end I just ignore the migration tool.

**Comment by p2** [comment]
Yeah sure! Xcode doesn't handle this well, until you switch to tabs in prefs it won't do the right thing. Let me know when you're done and I'll take another look!

**Comment by paulw11** [comment]
Hi

I think I have got them all now. 

Paul

File: Sources/Base/OAuth2Base.swift [hunk 1]
```diff
@@ -468,13 +468,13 @@ open class OAuth2ContextStore {
...
-		}
-		return _state
-	}
+    open var state: String {
+        if _state.isEmpty {
+            _state = UUID().uuidString
+            _state = String(_state[_state.startIndex..<_state.index(_state.startIndex, offsetBy: 8)])        // only use the first 8 chars, should be enough
+        }
+        return _state
+    }
```
Comment (p2) [inline Sources/Base/OAuth2Base.swift]: Spaces.

File: Sources/macOS/OAuth2WebViewController.swift [hunk 2]
```diff
@@ -216,7 +216,7 @@ public class OAuth2WebViewController: NSViewController, WKNavigationDelegate, NS
 	/**
 	Tells the web view to stop loading the current page, then calls the `onWillCancel` block if it has a value.
 	*/
-	func cancel(_ sender: AnyObject?) {
+    @objc func cancel(_ sender: AnyObject?) {
```
[the remark this answers is not in the document]
Comment (p2) [reply Sources/macOS/OAuth2WebViewController.swift]: Still spaces.

File: Sources/macOS/OAuth2WebViewController.swift [hunk 3]
```diff
@@ -280,8 +280,8 @@ public class OAuth2WebViewController: NSViewController, WKNavigationDelegate, NS
 	
 	
 	// MARK: - Window Delegate
-	
-	public func windowShouldClose(_ sender: Any) -> Bool {
+
```
[the remark this answers is not in the document]
Comment (p2) [reply Sources/macOS/OAuth2WebViewController.swift]: Still spaces.

File: Sources/Base/OAuth2Base.swift [hunk 4]
```diff
@@ -468,12 +468,13 @@ open class OAuth2ContextStore {
 	
 	We internally generate a UUID and use the first 8 chars if `_state` is empty.
 	*/
-	open var state: String {
+    open var state: String {
```
Comment (p2) [inline Sources/Base/OAuth2Base.swift]: Spaces.

File: Sources/Base/OAuth2Base.swift [hunk 5]
```diff
@@ -468,12 +468,13 @@ open class OAuth2ContextStore {
...
 		if _state.isEmpty {
-			_state = UUID().uuidString
-			_state = _state[_state.startIndex..<_state.index(_state.startIndex, offsetBy: 8)]		// only use the first 8 chars, should be enough
+            _state = UUID().uuidString
```
Comment (p2) [inline Sources/Base/OAuth2Base.swift]: Spaces.

File: Sources/Base/OAuth2Base.swift [hunk 6]
```diff
@@ -468,12 +468,13 @@ open class OAuth2ContextStore {
...
+			_state = String(_state[_state.startIndex..<_state.index(_state.startIndex, offsetBy: 8)])        // only use the first 8 chars, should be enough
 		}
 		return _state
+
```
Comment (p2) [inline Sources/Base/OAuth2Base.swift]: Extra newline.

File: Sources/macOS/OAuth2WebViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (p2) [reply Sources/macOS/OAuth2WebViewController.swift]: Still spaces.

**Comment by p2** [comment]
(Those spaces are on the newline before `public func`!

**Comment by paulw11** [comment]
https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/  :P

File: Sources/macOS/OAuth2WebViewController.swift [hunk 7]
```diff
@@ -280,8 +280,8 @@ public class OAuth2WebViewController: NSViewController, WKNavigationDelegate, NS
 	
 	
 	// MARK: - Window Delegate
-
```
Comment (p2) [inline Sources/macOS/OAuth2WebViewController.swift]: Now the indent is gone. 😁

**Comment by ksteigerwald** [comment]
When will the podspec be updated?

**Pull request merged**